### PR TITLE
Add appendix effort

### PR DESCRIPTION
**Pull request by ricketytoc**
Add appendix for effort.

### CODE REVIEW

**Review by Weiennn — APPROVED** [review]
LGTM

**Review by Chen-Kuei — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -968,3 +968,32 @@ The current implementation of the `sort` command allows users to input additiona
...
+1. As our salary attribute needed to hold a number and conduct operations on it, there were problems arising from the usage of floating point numbers. We had to change from using `double` to using `long` to fix those bugs.
+1. The implementation of the `sort` command was not straightforward. JavaFX API only has `SortedList` and `FilteredList`. To achieve both effects, we had to find a way to ‘combine’ them to allow sorting and filtering on the same list.
+1. For the `import` and `export` commands, the file path was difficult to handle. Different operating systems have different formats and specifications. The tests had to be designed for the different OS as well.
+1. For the `history` command, we had to decide where to store the class that was supposed to store the previous command and how it should be integrated into the application.
+1. For the ‘find’ command, we had to find a way to combine all the predicates into one so that we can find employees who fulfill multiple find parameters. There was also a challenge faced with finding employees by salary which is related to the aforementioned challenge with salary.
```
Comment (Chen-Kuei) [inline docs/DeveloperGuide.md:984]: 'find' command should be `find` command

Comment (ricketytoc) [reply docs/DeveloperGuide.md]: ok!

**Review by aexolate — APPROVED** [review]
LGTM

**Pull request merged**